### DISCUSSION TITLE
constraint_mode disabling constraint only for latest instance

### DISCUSSION THREAD

**Discussion by ShreyanJabade** [fvutils/pyvsc#79] [General]
Hello, 
Suppose we have multiple instances of a class as shown in snippet below. There are three instances of class `Child1` in the list `c1` inside the Parent class. It is intended to disable the constraint on only the first member of the list via `inst.c1[0].test_c.constraint_mode(False)`. However, the constraint on `c1[0]` is not disabled. It seems that the constraint is getting disabled only for the latest defined instance (in this case `c1[2]`).
I request you to look into this aspect.
Thank you

```
import vsc 
from vsc import *
@vsc.randobj
class Parent:
    def __init__(self):
        self.id = 0
        self.c1 = vsc.rand_list_t(vsc.rand_attr(Child1()))
        for i in range(3):    
            self.c1.append(vsc.rand_attr(Child1()))

@vsc.randobj
class Child1():
    def __init__(self):
        self.a = vsc.rand_uint8_t()
        self.b = vsc.rand_uint8_t()
    
    @vsc.constraint
    def test_c(self):
        self.a < 6
        self.b < 3
        self.a*self.b == 10

inst=Parent()
inst.c1[0].test_c.constraint_mode(False)
inst.randomize()

print("inst.c1[0].a =", inst.c1[0].a)
print("inst.c1[0].b =", inst.c1[0].b)
print("inst.c1[2].a =", inst.c1[2].a)
print("inst.c1[2].b =", inst.c1[2].b)

```
Output:
```
inst.c1[0].a = 5
inst.c1[0].b = 2
inst.c1[2].a = 0
inst.c1[2].b = 1
```

**Comment by mballance** [comment]
Hello @ShreyanJabade,
    Thank you for the testcase. Certainly simplified the process of tracking things down. In this case, the issue was related to how constraints are captured in Python. I've put a fix in place (0.4.1 release), and am able to show that the testcase works properly.

Best Regards,
Matthew

**Reply by ShreyanJabade** [reply]
Yes, this aspect is working as expected. Thank you for your consideration.
